src/invidious/videos.cr

```diff
@@ -877,42 +877,84 @@ def extract_video_info(video_id : String, proxy_region : String? = nil, context_
+    .try &.as_a.find(&.["videoSecondaryInfoRenderer"]?)
+      .try &.["videoSecondaryInfoRenderer"]
+
+  # Likes/dislikes
+
+  toplevel_buttons = video_primary_renderer
+    .try &.dig?("videoActions", "menuRenderer", "topLevelButtons")
+
+  if toplevel_buttons
+    likes_button = toplevel_buttons.as_a
+      .find(&.["toggleButtonRenderer"]["defaultIcon"]["iconType"].as_s.== "LIKE")
+      .try &.["toggleButtonRenderer"]
+
+    if likes_button
+      likes_txt = (likes_button["defaultText"]? || likes_button["toggledText"]?)
+        .try &.dig?("accessibility", "accessibilityData", "label")
+      likes = likes_txt.as_s.gsub(/\D/, "").to_i64? if likes_txt
+
+      LOGGER.trace("extract_video_info: Found \"likes\" button. Button text is \"#{likes_txt}\"")
+      LOGGER.debug("extract_video_info: Likes count is #{likes}") if likes
+    end
+
+    dislikes_button = toplevel_buttons.as_a
+      .find(&.["toggleButtonRenderer"]["defaultIcon"]["iconType"].as_s.== "DISLIKE")
+      .try &.["toggleButtonRenderer"]
+
+    if dislikes_button
+      dislikes_txt = (dislikes_button["defaultText"] || dislikes_button["toggledText"]?)
+        .try &.dig?("accessibility", "accessibilityData", "label")
+      dislikes = dislikes_txt.as_s.gsub(/\D/, "").to_i64? if dislikes_txt
+
+      LOGGER.trace("extract_video_info: Found \"dislikes\" button. Button text is \"#{dislikes_txt}\"")
+      LOGGER.debug("extract_video_info: Dislikes count is #{dislikes}") if dislikes
+    end
+  end
+
+  if likes && likes != 0_i64 && (!dislikes || dislikes == 0_i64)
+    if rating = player_response.dig?("videoDetails", "averageRating").try { |x| x.as_i64? || x.as_f? }
+      dislikes = (likes * ((5 - rating)/(rating - 1))).round.to_i64
+      LOGGER.debug("extract_video_info: Dislikes count (using fallback method) is #{dislikes}")
+    end
+  end
+
+  params["likes"] = JSON::Any.new(likes || 0_i64)
+  params["dislikes"] = JSON::Any.new(dislikes || 0_i64)
+
+  # Description
+
+  description_html = video_secondary_renderer.try &.dig?("description", "runs")
+    .try &.as_a.try { |t| content_to_comment_html(t).gsub("\n", "<br/>") }
+
+  params["descriptionHtml"] = JSON::Any.new(description_html || "<p></p>")
+
+  # Video metadata
+
+  metadata = video_secondary_renderer
+    .try &.dig?("metadataRowContainer", "metadataRowContainerRenderer", "rows")
+      .try &.as_a
 
   params["genre"] = params["microformat"]?.try &.["playerMicroformatRenderer"]?.try &.["category"]? || JSON::Any.new("")
   params["genreUrl"] = JSON::Any.new(nil)
 
   metadata.try &.each do |row|
     title = row["metadataRowRenderer"]?.try &.["title"]?.try &.["simpleText"]?.try &.as_s
-    contents = row["metadataRowRenderer"]?
-      .try &.["contents"]?
-        .try &.as_a[0]?
+    contents = row.dig?("metadataRowRenderer", "contents", 0)
 
     if title.try &.== "Category"
-      contents = contents.try &.["runs"]?
-        .try &.as_a[0]?
+      contents = contents.try &.["runs"]?.try &.as_a[0]?
```

Comment:
```suggestion
      contents = contents.try &.dig?("runs", 0)
```
I believe that's the same and fits with the theme of the PR

Reply:
Oh, the goal of this PR wasn't to rewrite/clean the whole code around, but that doesn't hurt ^^